hal/src/subghz/sx126x.rs

```diff
@@ -0,0 +1,252 @@
+use core::time::Duration;
+
+use radio::modulation::lora;
+use radio::modulation::lora::LoRaChannel;
+use radio::{BasicInfo, Busy, Receive, Transmit};
+
+use crate::spi::Spi3;
+use crate::subghz;
+use crate::subghz::rfs::{RfSwRx, RfSwTx};
+use crate::subghz::{
+    CalibrateImage, CfgIrq, CodingRate, FallbackMode, HeaderType, Irq, LoRaModParams,
+    LoRaPacketParams, LoRaSyncWord, Ocp, PaConfig, PacketType, RampTime, RegMode, RfFreq,
+    SpreadingFactor, StandbyClk, SubGhz, TcxoMode, TcxoTrim, Timeout, TxParams,
+};
+
+const IRQ_CFG: CfgIrq = CfgIrq::new()
+    .irq_enable_all(Irq::RxDone)
+    .irq_enable_all(Irq::Timeout)
+    .irq_enable_all(Irq::TxDone)
+    .irq_enable_all(Irq::Err);
+
+const PREAMBLE_LEN: u16 = 5 * 8;
+const TX_BUF_OFFSET: u8 = 0;
+const RX_BUF_OFFSET: u8 = 128;
+
+const PA_CONFIG: PaConfig = PaConfig::LP_10;
+const TX_PARAMS: TxParams = TxParams::LP_10.set_ramp_time(RampTime::Micros40);
+
+const TCXO_MODE: TcxoMode = TcxoMode::new()
+    .set_txco_trim(TcxoTrim::Volts1pt7)
+    .set_timeout(Timeout::from_millis_sat(10));
```

Comment:
I'll see if I can make this configurable! I could include these settings in the `Sx126x` struct. Would default settings be acceptable (with a builder pattern), or should the user specify all settings?

Reply:
> I'll see if I can make this configurable! I could include these settings in the Sx126x struct. Would default settings be acceptable (with a builder pattern), or should the user specify all settings?

I generally like to set things to sane defaults, and give the user the option to modify them.

One thing I try to do is make it so configuration can be made a `const`, because most applications will have a static configuration that doesn't need to be changed, and it is helpful to enable `const` for that usecase.

Reply:
I updated it, is this better? I wasn't sure what to do with `TX_BUF_OFFSET` and `RX_BUF_OFFSET`, are those configurable? Also not sure what the default `CalibrateImage` should be. For LoRaWAN, there is `EU863-870`, `US902-928`, `CN779-787`, `EU433`, etc.